pkg/olm/olm.go

```diff
@@ -203,30 +203,117 @@ func GenerateCSV(opConf *operator.Conf) *operv1.ClusterServiceVersion {
+				DisplayName:  "Target Bucket"},
+			operv1.SpecDescriptor{Description: "NumVolumes is the number of volumes to allocate.",
+				Path:         "pvPool.numVolumes",
+				XDescriptors: []string{uiFieldGroupPvPool, uiNumber},
+				DisplayName:  "Num Volumes"},
+			operv1.SpecDescriptor{Description: "VolumeResources represents the minimum resources each volume should have.",
+				Path:         "pvPool.resources",
+				XDescriptors: []string{uiFieldGroupPvPool, uiText},
+				DisplayName:  "Resources"},
+			operv1.SpecDescriptor{Description: "StorageClass is the name of the storage class to use for the PV's.",
+				Path:         "pvPool.storageClass",
+				XDescriptors: []string{uiFieldGroupPvPool, uiNumber},
+				DisplayName:  "Storage Class"},
+			operv1.SpecDescriptor{Description: "Endpoint is the S3 compatible endpoint: http(s)://host:port.",
+				Path:         "s3Compatible.endpoint",
+				XDescriptors: []string{uiFieldGroupS3Compatible, uiText},
+				DisplayName:  "End Point"},
+			operv1.SpecDescriptor{Description: "Secret refers to a secret that provides the credentials. The secret should define AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY.",
+				Path:         "s3Compatible.secret",
+				XDescriptors: []string{uiFieldGroupS3Compatible, uiK8sSecret},
+				DisplayName:  "Secret"},
+			operv1.SpecDescriptor{Description: "SignatureVersion specifies the client signature version to use when signing requests.",
+				Path:         "s3Compatible.signatureVersion",
+				XDescriptors: []string{uiFieldGroupS3Compatible, uiNumber},
+				DisplayName:  "Signature Version"},
+			operv1.SpecDescriptor{Description: "TargetBucket is the name of the target S3 bucket.",
+				Path:         "s3Compatible.targetBucket",
+				XDescriptors: []string{uiFieldGroupS3Compatible, uiText},
+				DisplayName:  "Target Bucket"},
+		},
+
+		"BucketClass": []operv1.SpecDescriptor{
+			operv1.SpecDescriptor{Description: "BackingStores is an unordered list of backing store names. The meaning of the list depends on the placement.",
+				Path:         "placementPolicy.tiers[0].backingStores[0]",
+				XDescriptors: []string{uiFieldGroupPlacementPolicy, uiText},
+				DisplayName:  "Backing Stores"},
+			operv1.SpecDescriptor{Description: "Placement specifies the type of placement for the tier If empty it should have a single backing store.",
+				Path:         "placementPolicy.tiers[0].placement",
+				XDescriptors: []string{uiFieldGroupPlacementPolicy, uiText},
+				DisplayName:  "Placement"},
 		},
-		"BackingStore":      []operv1.SpecDescriptor{},
-		"BucketClass":       []operv1.SpecDescriptor{},
 		"ObjectBucketClaim": []operv1.SpecDescriptor{},
 		"ObjectBucket":      []operv1.SpecDescriptor{},
 	}
 	for kind := range crdSpecDescriptors {
 		for i := range crdSpecDescriptors[kind] {
 			d := &crdSpecDescriptors[kind][i]
-			d.DisplayName = d.Path
-			d.Description = d.Path
+			d.DisplayName = d.DisplayName
+			d.Description = d.Description
```

Comment:
The reason I wrote this loop originally was that I was too lazy to actually fill `Description` and `DisplayName` for the fields, so I just copied `Path` to both. However this is now the case only for the `NooBaa` CRD and not for `BackingStore` and `BucketClass` which you added, and this assignment is not even doing anything as it is now.

So I just suggest to remove this loop altogether and instead fill those Description and DisplayName for the NooBaa CRD.

Reply:
Updated the code.